src/features/recovery/components/RecoveryEmail/VerifyEmail.tsx

```diff
@@ -23,45 +22,35 @@ export const NotVerifiedMessage = ({ onVerify }: { onVerify: () => void }) => {
   )
 }
 
-export const isNoContentResponse = (error: string) => {
-  return error === 'Invalid response content: No Content'
-}
-
 const VerifyEmail = ({ onCancel, onSuccess }: { onCancel: () => void; onSuccess: () => void }) => {
   const [error, setError] = useState<string>()
   const [verificationCode, setVerificationCode] = useState<string>('')
-  const { verifyEmailAddress, resendVerification } = useRecoveryEmail()
+  const { verifyEmailAddress, resendVerification, registerRecovery } = useRecoveryEmail()
   const dispatch = useAppDispatch()
 
   const handleRetry = async () => {
     try {
       await resendVerification()
       setError(undefined)
     } catch (e) {
-      console.log(e)
-      // TODO: logError
+      setError('Error requesting verification code. Please try again.')
     }
   }
 
   const handleVerify = async () => {
     try {
       setError(undefined)
-      const result = await verifyEmailAddress(verificationCode)
+      await verifyEmailAddress(verificationCode)
+      dispatch(
+        showNotification({
+          variant: 'success',
+          groupKey: 'verify-email-complete',
+          message: 'Your email address is verified',
+        }),
+      )
+      onSuccess()
+      await registerRecovery()
```

Comment:
We could also register the recovery one step sooner when the email is registered and not when its verified.